### PR TITLE
onepoll.php is reworked

### PR DESCRIPTION
**Pull request by annando**
Several database calls are replaced with the new calls, the config class is now used and some logic problems had been solved as well. (The variable ```$importer_uid``` was used before it was assigned)

### CODE REVIEW

File: include/onepoll.php [hunk 1]
```diff
@@ -330,47 +320,42 @@ function onepoll_run(&$argv, &$argc){
...
-			intval($importer_uid)
-		);
-		if (count($x) && count($mailconf)) {
-		    $mailbox = construct_mailbox_name($mailconf[0]);
+		$x = dba::select('user', array('prvkey'), array('uid' => $importer_uid), array('limit' => 1));
+
+		$condition = array("`server` != '' AND `uid` = ?", $importer_uid);
+		$mailconf = dba::select('mailacct', array(), $condition, array('limit' => 1));
+		if (dbm::is_result($x) && dbm::is_result($mailconf)) {
+		    $mailbox = construct_mailbox_name($mailconf);
```
Comment (MrPetovan) [inline include/onepoll.php]: Indentation issue here.

File: include/onepoll.php [hunk 2]
```diff
@@ -475,28 +457,28 @@ function onepoll_run(&$argv, &$argc){
...
 
 						// If it seems to be a reply but a header couldn't be found take the last message with matching subject
-						if (!x($datarray,'parent-uri') && $reply) {
+						if (!x($datarray, 'parent-uri') && $reply) {
 							$r = q("SELECT `uri` , `parent-uri` FROM `item` WHERE `title` = \"%s\" AND `uid` = %d AND `network` = '%s' ORDER BY `created` DESC LIMIT 1",
 								dbesc(protect_sprintf($datarray['title'])),
 								intval($importer_uid),
 								dbesc(NETWORK_MAIL));
 							if (dbm::is_result($r))
-								$datarray['parent-uri'] = $r[0]['parent-uri'];
```
Comment (MrPetovan) [inline include/onepoll.php]: Why did you remove the `[0]` here?

Comment (annando) [reply include/onepoll.php]: Ops
